### PR TITLE
Dependency injection rework

### PR DESCRIPTION
**Pull request by tobiasbalzano**
Changes to how Dependency Injection handles scopes due to Microsoft Dependency injection 
- Adds working with scopes to Microsoft Dependency injection
- Changes how to work with SimpleInjector dependency injection 
- Adds test for above

### CODE REVIEW

File: knightbus/src/KnightBus.Core/IDependencyInjection.cs [hunk 1]
```diff
@@ -4,12 +4,25 @@
...
+//{
+//    var service = scope.ServiceProvider.GetInstance<ITestService>();
+//}
+//
+//
+//
+//using(DependencyInjection.GetScope())
+//{
+//    var service = DependencyInjection.GetInstance<ITestService>();
+//}
```
Comment (BookBeatAdmin) [inline knightbus/src/KnightBus.Core/IDependencyInjection.cs:28]: Unused code

Comment (tobiasbalzano) [reply knightbus/src/KnightBus.Core/IDependencyInjection.cs]: right, fixed!

Comment (tobiasbalzano) [reply knightbus/src/KnightBus.Core/IDependencyInjection.cs]: Could only find reference to GetScope in JobExecutor

File: KnightBus.DependencyInjection.Tests.Unit/KnightBus.DependencyInjection.Tests.Unit.csproj [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <PackageReference Include="nunit" Version="3.12.0" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.15.1" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.4.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\knightbus-microsoft-dependencyinjection\src\KnightBus.Microsoft.DependencyInjection\KnightBus.Microsoft.DependencyInjection.csproj" />
+    <ProjectReference Include="..\knightbus-simpleinjector\src\KnightBus.SimpleInjector\KnightBus.SimpleInjector.csproj" />
+    <ProjectReference Include="..\knightbus\src\KnightBus.Core\KnightBus.Core.csproj" />
+    <ProjectReference Include="..\knightbus\src\KnightBus.Host\KnightBus.Host.csproj" />
```
Comment (BookBeatAdmin) [inline KnightBus.DependencyInjection.Tests.Unit/KnightBus.DependencyInjection.Tests.Unit.csproj:21]: These two references should not need to be explicity specified

**Pull request merged**